php/flights.php

```diff
@@ -141,71 +139,72 @@
 if (!$sth->execute($params)) {
     die('Error;Query ' . print_r($_GET, true) . ' caused database error ' . $sql . ', ' . $sth->errorInfo()[0]);
 }
-$first = true;
 
 if ($export == "export" || $export == "backup") {
     // Start with byte-order mark to try to clue Excel into realizing that this is UTF-8
     print "\xEF\xBB\xBFDate,From,To,Flight_Number,Airline,Distance,Duration,Seat,Seat_Type,Class,Reason,Plane,Registration,Trip,Note,From_OID,To_OID,Airline_OID,Plane_OID\r\n";
 }
-$gcmap_city_pairs = ''; // list of city pairs when doing gcmap export.
+
+/**
+ * @param $src mixed
+ * @param $dst mixed
+ * @param $flip bool
+ * @return array
+ */
+function flip($src, $dst, $flip) {
+    // if !$flip return [ $src, $dst ]
+    // if $flip return [ $dst, $src ]
+    return [
+        $flip ? $dst : $src,
+        $flip ? $src : $dst,
+    ];
+}
+
+$rows = [];
 foreach ($sth as $row) {
-    $note = $row["note"];
+    $toFlip = $row["opp"] == 'Y';
+    [$src_code, $dst_code] = flip(
+        $toFlip,
+        format_apcode2($row["src_iata"], $row["src_icao"]),
+        format_apcode2($row["dst_iata"], $row["dst_icao"])
+    );
+
+    if ($export == "gcmap") {
+        $rows[] = urlencode($src_code . '-' . $dst_code);
+        continue;
+    }
+
+    [$src_apid, $dst_apid] = flip($toFlip, $row["src_apid"], $row["dst_apid"]);
 
     if ($route) {
         $row["distance"] = gcPointDistance(
-            array("x" => $row["sx"], "y" => $row["sy"]),
-            array("x" => $row["dx"], "y" => $row["dy"])
+            ["x" => $row["sx"], "y" => $row["sy"]],
+            ["x" => $row["dx"], "y" => $row["dy"]]
         );
         $row["duration"] = gcDuration($row["distance"]);
         $row["code"] = $row["al_name"] . " (" . $row["code"] . ")";
-        $note = "";
-        if ($row["stops"] == "0") {
-            $note = "Direct";
-        } else {
-            $note = $row["stops"] . " stops";
-        }
+        $note = $row["stops"] == "0"
+            ? "Direct"
+            : $row["stops"] . " stops";
+
+        // TODO: This clobbers the $note set above... Suspect it should append?
         if ($row["codeshare"] == "Y") {
             $note = "Codeshare";
         }
-    }
-
-    if ($first) {
-        $first = false;
     } else {
-        if ($export == "export" || $export == "backup") {
-            printf("\r\n");
-        } elseif ($export != "gcmap") {
-            printf("\n");
-        }
+        $note = $row["note"];
```

Comment:
In theory, we don't need to do the `'' AS note` in the SELECT for `$route == true`